src/Misc/layoutroot/run-docker.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+function delay {
+    time=${1:-1}
+    if [ -x "$(command -v sleep)" ]; then
+        sleep $time >/dev/null
+    elif [ -x "$(command -v ping)" ]; then
+        ping -n $time 127.0.0.1 >nul
+    else
+        count=0
+
+        while [[ $count != $[$time*25000] ]]; do
+            echo "sleep" >/dev/null
+            count=$[$count+1]
+        done
+    fi
+}
+
+# Validate not sudo
+user_id=`id -u`
+if [ $user_id -eq 0 -a -z "$AGENT_ALLOW_RUNASROOT" ]; then
+    echo "Must not run interactively with sudo"
+    exit 1
+fi
+
+# Change directory to the script root directory
+# https://stackoverflow.com/questions/59895/getting-the-source-directory-of-a-bash-script-from-within
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+# Do not "cd $DIR". For localRun, the current directory is expected to be the repo location on disk.
+
+# Run
+shopt -s nocasematch
+
+# Determining if the "--once" flag was passed
+ONCE=false
+for a in $*; do
+    if [[ "$a" == "--once" ]]; then
+        ONCE=true
+    fi
+done
+
+FIRST_ARG=$1
+ARGUMENTS=$*
+
+function runAgent {
+    if [[ "$FIRST_ARG" == "localRun" ]]; then
+        "$DIR"/bin/Agent.Listener $ARGUMENTS
+    else
+        if [[ "$ONCE" = true ]]; then
+            "$DIR"/bin/Agent.Listener run $ARGUMENTS
+        else
+            echo "Starting Agent listener with startup type: service"
```

Comment:
I think this could lead to some confusion - since this is not a service run mode (comparing to interactive), but rather the parameter which basically just disables agent run as a separate process after self-update (which although is used in a service agent mode). Could it be better here like below?
"Starting Agent listener with startup type: service - to prevent running of an agent in a separate process after self-update."

Reply:
I agree, it will be better